components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/util/ActionExecutorConfig.java

```diff
@@ -73,6 +82,81 @@ public boolean isExecutionForActionTypeEnabled(ActionType actionType) {
         }
     }
 
+    /**
+     * Returns the HTTP client connection pool size based on the system configuration.
+     *
+     * @return The HTTP client connection pool size, or the default if the property is missing or invalid.
+     */
+    public int getHttpClientConnectionPoolSize() {
+
+        int poolSizePropertyValue = DEFAULT_HTTP_CLIENT_CONNECTION_POOL_SIZE_PROPERTY;
+        String poolSizeValue = (String) IdentityConfigParser.getInstance().getConfiguration().
+                get(HTTP_CLIENT_CONNECTION_POOL_SIZE_PROPERTY);
+        if (StringUtils.isNotBlank(poolSizeValue)) {
+            try {
+                poolSizePropertyValue = Integer.parseInt(poolSizeValue);
+            } catch (NumberFormatException e) {
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug("Failed to read Http client connection pool size property in identity.xml." +
+                            " Expects a number. Using the default value: " +
+                            DEFAULT_HTTP_CLIENT_CONNECTION_POOL_SIZE_PROPERTY, e);
+                }
```

Comment:
we don't need to check if debug logs are enabled unless we are doing some complex manipulation.

Reply:
Updated in https://github.com/wso2/carbon-identity-framework/pull/5935/commits/d3435deafaa2baea562aae75e85180c9a5838a7e